lib/key-handle-decorator.js

```diff
@@ -0,0 +1,64 @@
+/* @flow */
+
+import React from 'react';
+
+import KeyHandler from './key-handler';
+import {type KeyConstants} from './constants';
+import {matchesKeyboardEvent, eventKey} from './utils';
+import { type primitiveOrArray } from './types';
+
+export type KeyhandleDecoratorState = {|
+ keyValue?: ?primitiveOrArray<string>,
+ keyCode: ?primitiveOrArray<number>,
+ code: ?primitiveOrArray<string>,
+|};
+
+export type DecoratorProps = {|
+  keyValue: ?primitiveOrArray<string>,
+  keyCode: ?primitiveOrArray<number>,
+  code: ?primitiveOrArray<string>,
+  keyEventName?: KeyConstants,
+|}
+
+export default function keyHandleDecorator<T>(matcher?: typeof matchesKeyboardEvent): Function {
+  return (props?: DecoratorProps & T): Function => {
+    const { keyValue, keyCode, code, keyEventName, ...otherProps } = props || {};
+
+    return (Component) => (
+      class KeyHandleDecorator extends React.Component<T, KeyhandleDecoratorState> {
+        state: KeyhandleDecoratorState = {
+          keyCode: null,
+          keyValue: null,
+          code: null,
+        }
+
+        render() {
+          return (
+            <React.Fragment>
```

Comment:
We'll have to update our peerDependencies in package.json:

``` json
{
  "peerDependencies": {
    "react": ^16.2.0-0"
  }
}
```

Reply:
Fixed